test/server/graphql/v1/mutation.test.js

```diff
@@ -343,6 +356,54 @@ describe('server/graphql/v1/mutation', () => {
         });
       });
     });
+
+    describe('archives a collective', () => {
+      const archiveCollectiveMutation = gql`
+        mutation ArchiveCollective($id: Int!) {
+          archiveCollective(id: $id) {
+            id
+            isArchived
+          }
+        }
+      `;
+      const unarchiveCollectiveMutation = gql`
+        mutation UnarchiveCollective($id: Int!) {
+          unarchiveCollective(id: $id) {
+            id
+            isArchived
+          }
+        }
+      `;
+      it('fails if not authenticated', async () => {
+        const result = await utils.graphqlQuery(archiveCollectiveMutation, {
+          id: collective1.id,
+        });
+
+        expect(result.errors).to.exist;
+        expect(result.errors[0].message).to.equal('You need to be logged in to archive a collective');
+      });
+
+      it('fails if not authenticated as an Admin', async () => {
+        const result = await utils.graphqlQuery(archiveCollectiveMutation, { id: collective1.id }, user3);
+
+        expect(result.errors).to.exist;
+        expect(result.errors[0].message).to.equal('You need to be logged in as an Admin.');
+      });
+
+      it('should archive its children projects and events', async () => {
+        await utils.graphqlQuery(archiveCollectiveMutation, { id: collective1.id }, user1);
+
+        const projects = (await collective1.getProjects()).map(p => ({ id: p.id, name: p.name, isActive: p.isActive }));
+        const events = (await collective1.getEvents()).map(p => ({ id: p.id, name: p.name, isActive: p.isActive }));
+
+        expect(projects).to.all.have.property('isActive', false);
```

Comment:
I would make sure these arrays are not empty, in case someones edits the code used to seed the data

```suggestion
        expect(projects.length).to.eq(1);
        expect(events.length).to.eq(1);
        expect(projects).to.all.have.property('isActive', false);
```